premerge/advisor/advisor_lib.py

```diff
@@ -131,20 +132,57 @@ def _try_explain_failing_at_head(
     return None
 
 
+def _try_explain_flaky_failure(
```

Comment:
Maybe a comment describing the particular definition of "flaky" this function is using would be handy?

It looks to me like it's testing if this failure has been seen before over at least a 200 commit range between oldest/newest sighting? What's to stop that being "it's been failing at HEAD for at least 200 revisions"?

Reply:
Added a function level docstring.

I've never seen a non-flaky test failure that has been failing for more than ~20 commits. I think a OOM more in range should be safe. I thought about also adding a heuristic around the longest running subsequence to catch cases like this, but I think it adds complexity for little benefit:
1. As mentioned, 200 is about an OOM bigger than the ranges we've seen in practice.
2. We're exactly matching the failure message, so the test would need to fail in the exact same way to trigger this.